File: tests/machine/x/dart/group_by_left_join.dart
var customers = [{'id': 1, 'name': 'Alice'}, {'id': 2, 'name': 'Bob'}, {'id': 3, 'name': 'Charlie'}];

var orders = [{'id': 100, 'customerId': 1}, {'id': 101, 'customerId': 1}, {'id': 102, 'customerId': 2}];

var stats = (() {
  var _q0 = <dynamic>[];
  var _g1 = <dynamic, List<dynamic>>{};
  for (var c in customers) {
    var _jt2 = <dynamic>[];
    for (var o in orders) {
      if (!(o['customerId'] == c['id'])) continue;
      _jt2.add(o);
    }
    if (_jt2.isEmpty) _jt2.add(null);
    for (var o in _jt2) {
      var _k4 = c['name'];
      _g1.putIfAbsent(_k4, () => <dynamic>[]).add(c);
    }
  }
  for (var entry in _g1.entries) {
    var g = entry.value;
    var _k4 = entry.key;
    _q0.add({'name': _k4, 'count': (() {
  var _q5 = <dynamic>[];
  for (var r in g) {
    if (!(r['o'])) continue;
    _q5.add(r);
  }
  return _q5;
})().length});
  }
  return _q0;
})();

void main() {
  print('--- Group Left Join ---');
  var _iter6 = stats;
  for (var s in (_iter6 is Map ? (_iter6 as Map).keys : _iter6) as Iterable) {
    print([s['name'], 'orders:', s['count']].join(' '));
  }
}

File: tests/machine/x/c/group_by_left_join.c
#include <stdio.h>
#include <stdlib.h>

typedef struct {
  int id;
  char *name;
} CustomersItem;
typedef struct {
  int len;
  CustomersItem *data;
} list_CustomersItem;
static list_CustomersItem list_CustomersItem_create(int len) {
  list_CustomersItem l;
  l.len = len;
  l.data = calloc(len, sizeof(CustomersItem));
  if (!l.data && len > 0) {
    fprintf(stderr, "alloc failed\n");
    exit(1);
  }
  return l;
}

typedef struct {
  int id;
  int customerId;
} OrdersItem;
typedef struct {
  int len;
  OrdersItem *data;
} list_OrdersItem;
static list_OrdersItem list_OrdersItem_create(int len) {
  list_OrdersItem l;
  l.len = len;
  l.data = calloc(len, sizeof(OrdersItem));
  if (!l.data && len > 0) {
    fprintf(stderr, "alloc failed\n");
    exit(1);
  }
  return l;
}

typedef struct {
  char *name;
  int count;
} StatsItem;
typedef struct {
  int len;
  StatsItem *data;
} list_StatsItem;
static list_StatsItem list_StatsItem_create(int len) {
  list_StatsItem l;
  l.len = len;
  l.data = calloc(len, sizeof(StatsItem));
  if (!l.data && len > 0) {
    fprintf(stderr, "alloc failed\n");
    exit(1);
  }
  return l;
}

int main() {
  CustomersItem tmp1_data[] = {(CustomersItem){.id = 1, .name = "Alice"},
                               (CustomersItem){.id = 2, .name = "Bob"},
                               (CustomersItem){.id = 3, .name = "Charlie"}};
  list_CustomersItem tmp1 = {3, tmp1_data};
  list_CustomersItem customers = tmp1;
  OrdersItem tmp2_data[] = {(OrdersItem){.id = 100, .customerId = 1},
                            (OrdersItem){.id = 101, .customerId = 1},
                            (OrdersItem){.id = 102, .customerId = 2}};
  list_OrdersItem tmp2 = {3, tmp2_data};
  list_OrdersItem orders = tmp2;
  list_StatsItem stats = 0;
  printf("%s\n", "--- Group Left Join ---");
  // unsupported dynamic list iteration
  for (;;) {
    break;
  }
  return 0;
}
